package/tests/test_api/test_graphql.py

```diff
@@ -1,5 +1,186 @@
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
 from unittest import mock
-from unittest.mock import PropertyMock
+from unittest.mock import PropertyMock, call, patch
+from pathlib import Path
+
+import shutil
+import pytest
+from kedro.extras.datasets.pandas import CSVDataSet
+from kedro.extras.datasets.tracking import JSONDataSet, MetricsDataSet
+from kedro.io import DataCatalog, Version
+from strawberry import ID
+
+from kedro_viz.api.graphql import JSONObject, TrackingDataSet, get_run_tracking_data
+from kedro_viz.data_access.managers import DataAccessManager
+
+
+@pytest.fixture
+def example_tracking_output():
+    yield [
+        TrackingDataSet(
+            datasetName="metrics",
+            datasetType="kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet",
+            data={
+                "col1": [{"runId": "2021-11-02T18.24.24.379Z", "value": 1.0}],
+                "col2": [{"runId": "2021-11-02T18.24.24.379Z", "value": 2.0}],
+                "col3": [{"runId": "2021-11-02T18.24.24.379Z", "value": 3.0}],
+            },
+        ),
+        TrackingDataSet(
+            datasetName="more_metrics",
+            datasetType="kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet",
+            data={
+                "col4": [{"runId": "2021-11-02T18.24.24.379Z", "value": 4.0}],
+                "col5": [{"runId": "2021-11-02T18.24.24.379Z", "value": 5.0}],
+                "col6": [{"runId": "2021-11-02T18.24.24.379Z", "value": 6.0}],
+            },
+        ),
+        TrackingDataSet(
+            datasetName="json_tracking",
+            datasetType="kedro.extras.datasets.tracking.json_dataset.JSONDataSet",
+            data={
+                "col2": [{"runId": "2021-11-02T18.24.24.379Z", "value": True}],
+                "col3": [{"runId": "2021-11-02T18.24.24.379Z", "value": 3}],
+                "col7": [
+                    {"runId": "2021-11-02T18.24.24.379Z", "value": "column_seven"}
+                ],
+            },
+        ),
+    ]
+
+
+class TestTrackingData:
+    def test_graphql_run_tracking_data_query(
+        self, tmp_path, data_access_manager: DataAccessManager, example_tracking_output
+    ):
+        save_version = "2021-11-02T18.24.24.379Z"
```

Comment:
I feel like this string is repeated, so it could be a fixture. This fixture could be used in `example_tracking_output` as well.